core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1456,10 +1456,8 @@ public ParsedSchema parseSchema(
       Throwable cause = e.getCause();
       if (cause instanceof InvalidSchemaException) {
         throw (InvalidSchemaException) cause;
-      } else if (cause instanceof RuntimeException) {
-        throw (RuntimeException) cause;
       } else {
-        throw new RuntimeException(e);
+        throw new InvalidSchemaException(e);
```

Comment:
are we always throwing InvalidSchemaException now?

Reply:
Yes, otherwise a 500 can result